hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -336,11 +337,11 @@ public void refreshTimeline() throws IOException {
           //if previous checkpoint is an empty string, skip resume use Option.empty()
           resumeCheckpointStr = Option.of(commitMetadata.getMetadata(CHECKPOINT_KEY));
         } else if (commitMetadata.getOperationType() == WriteOperationType.CLUSTER) {
-          // incase of CLUSTER commit, no checkpoint will be available in metadata.
+          // in case of CLUSTER commit or sql source, no checkpoint will be available in metadata.
           resumeCheckpointStr = Option.empty();
         } else if (HoodieTimeline.compareTimestamps(HoodieTimeline.FULL_BOOTSTRAP_INSTANT_TS,
             HoodieTimeline.LESSER_THAN, lastCommit.get().getTimestamp())) {
-          throw new HoodieDeltaStreamerException(
+          LOG.warn(
```

Comment:
This changes the behavior for all the sources. In some cases it might make sense to throw an exception instead of silently failing. Do you think it would be better to introduce a config `hoodie.deltastreamer.fail.on.missing.checkpoint` which is by default true? For SqlSource, users can set to false.